<!doctype html>
<html><head><title>VarSelect Manual</title><meta charset="UTF-8"><link type="text/css" rel="stylesheet" href="http://fonts.googleapis.com/css?family=Crimson+Text:400,400italic,700,700italic|Roboto:400,700,700italic,400italic"><style>/*
 * Copyright 2014 Quip
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License. You may obtain
 * a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

body {
    font-size: 15px;
    color: #333;
    background: #fff;
    padding: 60px 95px;
/*    max-width: 900px; */
    margin: 0 auto;
    text-rendering: optimizeLegibility;
    font-feature-settings: "kern";
    font-kerning: normal;
    -moz-font-feature-settings: "kern";
    -webkit-font-feature-settings: "kern";
}

/* Headings */
h1, h2, h3, th {
    font-family: Roboto, sans-serif;
    font-weight: 700;
    margin: 0;
    margin-top: 1.25em;
    margin-bottom: 0.75em;
}

h1 {
    font-size: 35px;
    line-height: 42px;
}

h1:first-child {
    margin-top: 0;
}

h2 {
    font-size: 18px;
    line-height: 22px;
}

h3 {
    text-transform: uppercase;
    font-size: 13px;
    line-height: 16px;
}

/* Body text */
body, p, ul, ol, td {
    font-family: 'Crimson Text', serif;
    font-size: 16px;
    line-height: 20px;
}

blockquote, q {
    display: block;
    margin: 1em 0;
    font-style: italic;
}

blockquote a, q a {
    text-decoration: underline;
}

blockquote {
    padding-left: 10px;
    border-left: 4px solid #a6a6a6;
}

q {
    color: #a6a6a6;
    line-height: 40px;
    font-size: 24px;
    text-align: center;
    quotes: none;
}

q a {
    color: #a6a6a6;
}

code, pre {
    font-family: Consolas, "Liberation Mono", Menlo, "Courier Prime Web", Courier, monospace;
    background: #f3f3f3;
}

code {
    padding: 1px;
    margin: 0 -1px;
    border-radius: 3px;
}

pre {
    display: block;
    line-height: 20px;
    text-shadow: 0 1px white;
    padding: 5px 5px 5px 30px;
    white-space: nowrap;
    position: relative;
    margin: 1em 0;
}

pre:before {
    content: "";
    position: absolute;
    top: 0;
    bottom: 0;
    left: 15px;
    border-left: solid 1px #dadada;
}

/* Lists */
div[data-section-style="5"],
div[data-section-style="6"],
div[data-section-style="7"] {
    margin: 12px 0;
}

ul {
    padding: 0 0 0 40px;
}

ul li {
    margin-bottom: 0.4em;
}

/* Bulleted list */
div[data-section-style="5"] ul {
    list-style-type: disc;
}
div[data-section-style="5"] ul ul {
    list-style-type: circle;
}
div[data-section-style="5"] ul ul ul {
    list-style-type: square;
}
div[data-section-style="5"] ul ul ul ul {
    list-style-type: disc;
}
div[data-section-style="5"] ul ul ul ul ul {
    list-style-type: circle;
}
div[data-section-style="5"] ul ul ul ul ul ul {
    list-style-type: square;
}

/* Numbered list */
div[data-section-style="6"] ul {
    list-style-type: decimal;
}
div[data-section-style="6"] ul ul {
    list-style-type: lower-alpha;
}
div[data-section-style="6"] ul ul ul {
    list-style-type: lower-roman;
}
div[data-section-style="6"] ul ul ul ul {
    list-style-type: decimal;
}
div[data-section-style="6"] ul ul ul ul ul {
    list-style-type: lower-alpha;
}
div[data-section-style="6"] ul ul ul ul ul ul {
    list-style-type: lower-roman;
}

/* Checklist */
div[data-section-style="7"] ul {
    list-style-type: none;
}

div[data-section-style="7"] ul li:before {
    content: "\2610";
    position: absolute;
    display: inline;
    margin-right: 1.2em;
    margin-left: -1.2em;
}

div[data-section-style="7"] ul li.parent:before {
    content: "";
}

div[data-section-style="7"] ul li.parent {
    font-weight: bold;
}

div[data-section-style="7"] ul li.checked {
    text-decoration: line-through;
}

div[data-section-style="7"] ul li.checked:before {
    content: "\2611";
    text-decoration: none;
}

/* Tables */
div[data-section-style="8"] {
    margin: 12px 0;
}

table {
    border-spacing: 0;
    border-collapse: separate;
    border: solid 1px #7c7c7c;
    box-shadow: 0 1px 2px rgba(0, 0, 0, .25);
    table-layout: fixed;
    position: relative;
}

table th, table td {
    padding: 2px 2px 0;
    min-width: 1.5em;
    word-wrap: break-word;
}

table th {
    border-bottom: 1px solid #ccc;
    background: #f0f0f0;
    font-weight: bold;
    vertical-align: bottom;
    color: #3a4449;
    text-align: center;
}

table td {
    padding-top: 0;
    border-left: 1px solid #e1e1e1;
    border-top: 1px solid #e1e1e1;
    vertical-align: top;
}

table td.bold {
    font-weight: bold;
}

table td.italic {
    font-style: italic;
}

table td.underline {
    text-decoration: underline;
}

table td.strikethrough {
    text-decoration: line-through;
}

table td.underline.strikethrough {
    text-decoration: underline line-through;
}

table td:first-child {
    border-left: hidden;
}

table tr:first-child td {
    border-top: hidden;
}

/* Images */
div[data-section-style="11"] {
    margin-top: 20px;
    margin-bottom: 20px;
    margin-left: auto;
    margin-right: auto;
}

div[data-section-style="11"][data-section-float="0"] {
    clear: both;
    text-align: center;
}

div[data-section-style="11"][data-section-float="1"] {
    float: left;
    clear: left;
    margin-right: 20px;
}

div[data-section-style="11"][data-section-float="2"] {
    float: right;
    clear: right;
    margin-left: 20px;
}

div[data-section-style="11"] img {
    display: block;
    max-width: 100%;
    height: auto;
    margin: auto;
}

hr {
    width: 70px;
    margin: 20px auto;
}
</style></head><body><h1 id='HZJACAGe9Ys'>VarSelect Manual</h1>

<!--<h2 id='HZJACAzLZji'><b>Manual</b></h2>-->
<div style="float:left; width:300px; font-family: sans-serif;">
    <ul style="list-style-type: none;">
	<li style="margin:6px;"><a href="#HZJACAI24Il" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Before you start</a></li>
	<li style="margin:6px;"><a href="#HZJACAYfkDS" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Download VarSelect</a></li>
	<li style="margin:6px;"><a href="#HZJACA2f7gw" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Install VarSelect</a></li>
	<li style="margin:6px;"><a href="#HZJACADqJSI" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Quick Start</a></li>
	<li style="margin:6px;"><a href="#HZJACAhTr2s" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Built-in analytic workflow for family and paired sample analysis</b></a></li>
	<li style="margin:6px;"><a href="#HZJACAaoTrN" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Start from vcf files by multiple callers </b></a></li>
	<li style="margin:6px;"><a href="#HZJACAShXyB" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Original analysis and updates of VarSelect database </b></a></li>
	<li style="margin:6px;"><a href="#HZJACAKTNXZ" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Comparative analysis: compare results from any two original analyses </b></a></li>
	<li style="margin:6px;"><a href="#HZJACAxVuTT" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Description of VarSelect scripts</a></li>
	<li style="margin:6px;"><a href="#HZJACA2d2cY" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Examples</b></a></li>
	<li style="margin:6px;"><a href="#HZJACAI5rli" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Example 1 - samples of a family study</b></a></li>
	<li style="margin:6px;"><a href="#HZJACAA6Fli" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Example 2 - paired case/control samples</a></li>
	<li style="margin:6px;"><a href="#HZJACAC8Toc" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Example 3 – Re-analysis and comparison (comparative analysis)</b></a></li>
	<li style="margin:6px;"><a href="#HZJACAKggXg" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Example 4 – comparison of multiple variants callers</b></a></li>
	<li style="margin:6px;"><a href="#HZJACACRuNn" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Known issues</a></li>
	<li style="margin:6px;"><a href="#HZJACAFpRLy" style="text-decoration:none;font-face:light;font-family: sans-serif; ">Reference</b></a></li>	
    </ul>
</div>


<div style="max-width:800px;float:left;">
<img src="fig1.jpg" width="800px"/>
<h2 id='HZJACAI24Il'><b>Before you start</b></h2>

The following tools are required for running VarSelect, please follow the instruction and install all the following tools for full benefits. Please note that the current version of VarSelect supports only the human genome reference hg19/GRCh37. The support for GRCh38 will be available shortly. <br><br><b>Gemini</b><br> Gemini[1] is an analytic framework for human genetic variations by taking advantage of the SQLite database engine. It can be downloaded and installed by following the instruction at <a href="https://gemini.readthedocs.io/en/latest/content/installation.html"><code>https://gemini.readthedocs.io/en/latest/content/installation.html</code></a>. Once installed, please follow the commands to download the data files of the additional annotations, such as the GERP[2] and CADD[3] scores at <a href="https://gemini.readthedocs.io/en/latest/content/installation.html#updating-the-gemini-executables-and-annotations"><code>https://gemini.readthedocs.io/en/latest/content/installation.html#updating-the-gemini-executables-and-annotations</code></a>.<br><br><b>VEP</b><br> VEP[4] is a comprehensive variants annotation tool, and is part of the Ensembl project. To install VEP, please follow the instruction at <a href="http://www.ensembl.org/info/docs/tools/vep/script/vep_tutorial.html"><code>http://www.ensembl.org/info/docs/tools/vep/script/vep_tutorial.html</code></a>. Please note that the pre-built cache files are required for speeding up the annotation process. Please follow the instruction step by step at <a href="http://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html"><code>http://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html</code></a> to download VEP of GRCh37 version. <br><br><b>VEP plugins</b><br> VEP supports plugin modules to incorporate annotations from the external datasets. VarSelect requires the dbNSFP plugin, which are available at <a href="https://github.com/Ensembl/VEP_plugins/blob/release/86/dbNSFP.pm"><code>https://github.com/Ensembl/VEP_plugins/blob/release/86/dbNSFP.pm</code></a>, respectively. <br><br><b>dbNSFP</b><br> dbNSFP[5, 6] annotates genome-wide non-synonymous single nucleotide variations (nsSNVs). The data file of dbNSFP is available at <a href="https://sites.google.com/site/jpopgen/dbNSFP"><code>https://sites.google.com/site/jpopgen/dbNSFP</code></a>. <br><br><b><b>annovar</b></b><br> ANNOVAR[7] is a variant annotation tool with high efficiency to a variety of annotation databases, and is available at <a href="http://annovar.openbioinformatics.org/en/latest/user-guide/download/"><code>http://annovar.openbioinformatics.org/en/latest/user-guide/download/</code></a>. Please note that a license is required. Please follow the instruction at <a href="http://annovar.openbioinformatics.org/en/latest/user-guide/startup/"><code>http://annovar.openbioinformatics.org/en/latest/user-guide/startup/</code></a> to install scripts into proper directories when all the required packages are downloaded. Databases will be automatically installed by the VarSelect installation script.<br><br><b>snpEff</b><br> snpEff[8] annotates and predicts the impact of genetic variants, and is available at <a href="http://snpeff.sourceforge.net/download.html"><code>http://snpeff.sourceforge.net/download.html</code></a>. After downloading, please also download the pre-built snpEff annotation database. Please note that the current version of VarSelect is based on human genome reference hg19/GRCh37. Please download it with following command:<br/>

<i><i>java –jar /path/to/your/snpEff.jar download –v GRCh37.75</i></i><br/>

 <br/>

<b><b>vcftools</b></b><br/>

<a href="https://vcftools.github.io/"><code>vcftools</code></a>[9] is a set of tools for manipulating genetic variants in the VCF-formatted files, and is available at <a href="https://vcftools.github.io/index.html"><code>https://vcftools.github.io/index.html</code></a>. Please follow the instruction to install vcftools at <a href="https://vcftools.github.io/examples.html"><code>https://vcftools.github.io/examples.html</code></a>.<br><br><b>bcftools, bgzip, tabix</b><br> bcftools, bgzip and tabix are tools to compress, index and manipulate VCF files. bcftools is available at <a href="http://www.htslib.org/download/"><code>http://www.htslib.org/download/</code></a>, and includes the bgzip and the tabix tools in the software release.<br/>

<h2 id='HZJACAYfkDS'><b>Download VarSelect</b></h2>

The latest version of VarSelect is available a<i>t</i>  <a href="https://github.com/VarSelect/">https://github.com/VarSelect/</a>  <i><i>.</i></i><br/>

<h2 id='HZJACA2f7gw'><b>Install VarSelect</b></h2>

Please make sure that you have downloaded all the required packages and resources needed by VarSelect. When you are all set, please run the following command to decompress the VarSelect files.<br/>

<pre id='HZJACAlwcQZ'>        <i><i>tar zxvf VarSelect-latest.tar.gz</i></i></pre>

After extracting the package, run the VarSelect installation script<br/>

<pre id='HZJACAPze2x'>        <i>/path/to/your/VarSelect/install_VarSelect.pl</i></pre>

Add the VarSelect path to your system's $PATH settings<br/>

<pre id='HZJACA7ZIqe'>        <i><i>export PATH=/path/to/your/VarSelect/dir:$PATH</i></i></pre>

 <br/>

<h2 id='HZJACADqJSI'><b>Quick Start</b></h2>

VarSelect script is executable only on command line. Please use ‘-h’ flag for the basic usage information. <br/>

<pre id='HZJACAVzUZR'><code>varselect.pl -h</code></pre>

VarSelect annotates and analyzes sequence variants, and compares the results from different original analyses. To start using VarSelect, please use ‘annotate’ to process your vcf file(s) of interests.<br/>

<pre id='HZJACA2dBku'>varselect.pl annotate -v /path/to/vcf_files_list<br>                      -p /path/to/ped/file <br>                      -m workflow_mode</pre>

The annotation function combines the variants from different samples and annotates all reported variants of 406 features. The ‘-v’ option specifies the file describing the links between samples and corresponding variant files. The link is specified by a comma separator, one file per line as the following format:<br/>

<pre id='HZJACAPEWtg'>sample1,/path/to/vcf/file1<br> sample1,/path/to/vcf/file2<br> sample2,/path/to/vcf/file3 </pre>

The gender and phenotype information is also required, and is specified by the ‘-p’ option followed by a PED file containing the pedigree information. An example of the PED file is available at http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped.<br/>

 <br/>

When the variants are annotated, VarSelect creates a SQLite database file by using the <code>GEMINI framework.</code><code>[1]</code> The database can be queried and filtered by using the GEMINI and/or the SQL commands to extract the annotation information.<br/>

<pre id='HZJACA7yTmi'>gemini query --header -q 'select * from variants limit 10' /path/to/varselect.db</pre>

VarSelect also annotates copy number variation (CNV) produced by CNVkit[10] to the variants. The option -c specifies the file describing the list of CNV calls as following format:<br/>

<pre id='HZJACA30zA9'>sample1,/path/to/cns/file1<br> sample2,/path/to/cns/file2</pre>

VarSelect annotates CNV to the database with tags ‘cnv_samples’ and ‘cnv_log2’, and computes quantitative change by log base 2 between the paired samples. The results are tagged ‘cnv_fc_samples’ and ‘cnv_foldchange_log2’. <br/>

 <br/>

VarSelect also integrates the gene expression profiles, if provided, to annotate sequence variants. VarSelect currently supports the results from DEXseq,[11, 12] sailfish[13] and featureCounts.[14] VarSelect annotates the variants with the following tags ‘xpr_samples’, ‘xpr_readcounts’, ‘xpr_fpkm’, and ‘xpr_tpm’. The option ‘–x’ specifies the file describing the list of gene expression profiles as following format:<br/>

<pre id='HZJACACQLOh'>sample1,readcount,/path/to/file/of/reads_counts<br> sample1,exoncount,/path/to/file/of/dexseq_generated_exoncounts<br> sample1,fpkm,/path/to/combined.fpkm<br> sample1,tpm,/path/to/combined.gene.sf.tpm<br> sample2,readcount,/path/to/file/of/reads_counts<br> sample2,exoncount,/path/to/file/of/dexseq_generated_exoncounts<br> sample2,fpkm,/path/to/combined.fpkm<br> sample2,tpm,/path/to/combined.gene.sf.tpm</pre>

 <br/>

Quantitative change between the paired samples is pairwise computed and annotated with the tags ‘xpr_fc_samples’, ‘xpr_foldchange_readcount’, ‘xpr_foldchange_fpkm’, and ‘xpr_foldchange_tpm’.<br/>

<br/>

<h2 id='HZJACAhTr2s'><b><b>Built-in analytic workflow for family and paired sample analysis</b></b></h2>

In the original analysis, VarSelect provides two common analytic workflows: 1) family and 2) paired case-control analysis, by specifying the option ‘-m family’ or ‘-m paired’, respectively.<br/>

 <br/>

For family analysis, VarSelect analyzes the genetic variants for five genetic models, including autosomal recessive (AR), compound heterozygosity (CH), de novo recessive (DNR), second-hit recessive (SHR) and X-link recessive (XLR). The variants satisfied the criteria of each of the models will be labelled with the tags: ‘is_AR’, ‘is_CH’, ‘is_DR’, ‘is_SH’ and ‘is_XL’, respectively.<br/>

 <br/>

For paired case-control analysis, VarSelect labels the changes of nucleotides between the paired samples with either 'loss of heterozygous (LOH)' or 'de novo mutations'. The changes are labelled with the tags: ‘is_LOH’ and ‘is_denovo’. The result of each analysis is recorded by a tag ‘in_analysis_{analysis id}’, while the 'analysis id' is the time when the analysis begins.<br/>

<h2 id='HZJACAaoTrN'><b><b>Start from vcf files by multiple callers </b></b></h2>

Different variant callers deliver inconsistent variant calling reports while the majority are consistent.[15] VarSelect deals with such situation by processing multiple VCF files by different variants callers in two ways: 1) unify all reported variants or 2) intersect variants reported by all callers. The option ‘-k’ flag triggers this function, followed by the options ‘-u’ (union) or ‘-i’ (intersection), depending on the analytic purpose.<br/>

<pre id='HZJACA6UDxl'>varselect.pl annotate -v /path/to/vcf/files/list <br>                      -p /path/to/ped/file <br>                      -m workflow_mode <br>                      -k  -i</pre>

 <br/>

The list specifying VCF files is slightly different from the single-caller mode since a sample would now have multiple VCF files. User must specify sample, variant callers, and the associated VCF files. An extra comma separated field with the name of variant caller is added to the end of each line as follows:<br/>

<pre id='HZJACAl4lhu'>sample1,/path/to/vcf/file1,<b>caller1</b><br> sample1,/path/to/vcf/file2,<b>caller2</b><br> sample2,/path/to/vcf/file3,<b>caller1</b><br> sample2,/path/to/vcf/file4,<b>caller2</b></pre>

Please note that regardless selection for the union or intersection, inconsistent calls among different callers are regarded ambiguity and are marked and removed from further analysis. The list of removed variants will also be stored in the result directory.<br/>

<h2 id='HZJACAShXyB'><b><b>Original analysis and updates of VarSelect database </b></b></h2>

Samples can go through either the family or the paired case-control workflow (namely original analysis) depending on the study design. Original analysis can be performed repeatedly by inclusion and/or exclusion of samples. The label of the phenotypic information (e.g. tumor and normal; affected and unaffected) of the samples can be changed and thereafter analyzed, according to the labels specified in the PED file. Samples begun with a hash character '#' in the PED file are excluded from the downstream analysis. In the example of the PED file shown here, the samples ‘uncle’ and ‘aunt’ are herein excluded from the downstream analysis. <br/>

<pre id='HZJACAH7nT2'>#family_id  sample_id    paternal_id     maternal_id     sex   phenotype<br> family1       father          0              0            1        1<br> family1       mother          0              0            2        1<br> family1       daughter      father         mother         2        2<br> #family1       uncle           0              0            2        2<br> #family1       aunt            0              0            2        2</pre>

 <br/>

A first full run of original analysis creates a VarSelect database, which is required for recording all the results of subsequent re-analyses (repetitive original analyses). The ‘-d’ option specifies the location of where the database file is stored. <br/>

<pre id='HZJACAdrEtC'>varselect.pl analysis  -d /path/to/gemini/db <br>                       -p /path/to/modified/ped/file <br>                       -m &lt;workflow mode&gt;<br>                       -k  -u</pre>

After re-analysis, a new analysis directory will be created named with a new analysis id. The results and logs will also be stored in new directory. Filtered variants will be assigned new tag in_analysis_{new analysis id} in the VarSelect database.<br/>

<h2 id='HZJACAKTNXZ'><b><b>Comparative analysis: compare results from any two original analyses </b></b></h2>

The results from different original analyses can be compared for specific purposes and is termed ‘comparative analysis’. Comparison between any two original analyses is specified by the options ‘-a’ and ‘-b’ as follows. <br/>

<pre id='HZJACAp0rBC'>varselect.pl compare  -a &lt;Analysis id of analysisA&gt; <br>                      -b &lt;Analysis id of analysisB&gt;<br>                      -c [1-4]    <br>                      -d /path/to/db_file</pre>

<i><i> </i></i><br/>

The comparative analysis includes four comparisons including: 1) the union of analysis A and B. 2) intersection of analysis A and B. 3) variants presented in the analysis A but not in the analysis B. 4) variants present in the analysis B but not in the analysis A. Results of new comparative analysis will be stored in a new directory. The filtered variants will be assigned a tag in_analysis with the analysis id.<br/>

 <br/>

<h2 id='HZJACAxVuTT'><b>Description of VarSelect scripts</b></h2>

<p id='HZJACA8yMrL'>Bellows are the short description of each VarSelect script. </p>

<div data-section-style='5'><ul id='HZJACAmQ5G4'><ul><li id='HZJACAwXW03' class='parent' value='1'><b>varselect.pl</b> is the main script of VarSelect, and includes three commands: annotate (initial annotation), analysis (original and re-analysis) and compare (comparative analysis). 

<br/></li><ul><li id='HZJACAoRgIG' class=''>Command ‘annotate’ triggers the script vs_annotate.pl to annotate VCF files from scratch. It also triggers the script vs_analysis.pl to specify workflow of choice. There are three required options: ‘-v sample-vcf file list’, ‘-p PED file’, ‘-m workflow mode’. 

<br/></li><li id='HZJACA2KKxl' class=''>Command ‘analysis’ triggers the script vs_analysis.pl. There are three required options: ‘-d gemini db file’, ‘-p PED file’, ‘-m workflow mode’. 

<br/></li><li id='HZJACAIbZxZ' class=''>Command ‘compare’ triggers the script vs_compare.pl to compare results between two or multiple original analyses. There are four required options: ‘-a’ and ‘-b’ to specify the id of the analysis A and B. Option ‘-c’ specifies the method of comparison (1. union, 2. intersection, 3. A only, and 4. B only). Option ‘-d’ specifies the VarSelect database file.

<br/></li></ul><li id='HZJACAvJRlf' class='parent'><b>vs_annotate.pl</b> is triggered by varselect.pl for three functions. 

<br/></li><ul><li id='HZJACAfwOS4' class=''>Firstly, it processes the VCF files, of which from same sample will be joined together by vcf-concat of VCFtools.[9] VCF files of different samples are then merged into a single VCF file by vcf-merge included in the VCFtools. The variants in the sex chromosome in merged VCF file will be fixed for ploidy by vcf-fix-ploidy in the VCFtools.

<br/></li><li id='HZJACAZZtJu' class=''>Secondly, the script triggers VEP,[4] snpEff[8] and ANNOVAR[7] for annotation.

<br/></li><li id='HZJACAmU833' class=''>Thirdly, the script triggers the Gemini framework to generate a SQLite database for downstream analysis. 

<br/></li></ul><li id='HZJACAkevsf' class='parent'><b>vs_analysis.pl</b> is triggered by varselect.pl to analyze variants by taking into account the genotypes, disease status, and the provided copy number and gene expression information. 

<br/></li><ul><li id='HZJACAGgCCF' class=''>If option ‘-k’ specified, multi-caller mode is turned on with union (option ‘-u’) or intersection (option ‘-i’) set of variants from different callers.

<br/></li><li id='HZJACAe5DYj' class=''>If option ‘-c’ specified, the script triggers cnvkit_parse.pl to annotate copy number variation to the variants.

<br/></li><li id='HZJACAgkiap' class=''>If option ‘–x’ specified, the script triggers xprprofile_parse.pl to annotate gene expression information to the variants.

<br/></li><li id='HZJACA4WnDH' class=''>The script supports two built-in analytic workflows by specifying -m option. For the paired case-control workflow, it triggers loh_detector.pl and denovo_detector.pl scripts to classify variants of loss of heterozygosity or of de novo changes. For the family workflow, the script triggers five scripts including Autosomal-recessive.py, Compound-het.py, Denovo-recessive.py, Second-hit.py and X-linked.py to classify the variants into the related genetic models. All variants selected by the workflow will be annotated on the tag ‘in_analysis_{analysis id}’ in the varselect_variants table of the database.

<br/></li></ul><li id='HZJACAhbzNI' class=''><b>vs_compare.pl</b> is triggered by varselect.pl to extract the intersection, union, or subtraction of variants between any two original analyses.

<br/></li><li id='HZJACAOuj5e' class=''><b>vsl_annot.pl</b> is triggered by vs_annotation.pl, vs_analysis.pl and vs_compare.pl to add new columns and that attaching annotation in the VarSelect database.

<br/></li><li id='HZJACAdWBx8' class=''><b>create_view_for_varselect.pl</b> is triggered by vs_annotation.pl, vs_analysis.pl and vs_compare.pl to create a view by merging the updated tables in the VarSelect database file.

<br/></li><li id='HZJACACd5Z9' class=''><b>run_vep.pl</b> is triggered by vs_annotate.pl to annotate VCF file through the Ensembl VEP.[4] The script also enables the dbNSFP plugin[5, 6] to annotate the non-synonymous variants.

<br/></li><li id='HZJACA2CGPD' class=''><b>run_snpeff.pl</b> is triggered by vs_annotate.pl to annotate VCF file through snpEff.[8] 

<br/></li><li id='HZJACA5ONFb' class=''><b>run_annovar.pl</b> is triggered by vs_annotate.pl to annotate VCF file through ANNOVAR.[7]

<br/></li><li id='HZJACA55nVl' class=''><b>GO_parse.pl</b> is triggered by vs_annotate.pl to annotate variants with the Gene Ontology (GO) terms.[16] The GO terms are assigned based on the transcripts where variants reside.

<br/></li><li id='HZJACAs4fp7' class=''><b>pathway_parse.pl</b> is triggered by vs_annotate.pl to annotate variants on KEGG pathways.[17-19] Associated pathways will be extracted by the built-in ‘gemini pathways’ in the GEMINI framework.

<br/></li><li id='HZJACAjbjcX' class=''><b>cnvkit_parser.pl</b> is triggered by vs_analysis.pl to annotate variants with the information of copy number variation. It extracts the log2 values where each variant resides and compute the CNV changes between the paired samples.

<br/></li><li id='HZJACAcbpSk' class=''><b>xprprofile_parser.pl</b> is triggered by vs_analysis.pl to annotate variants with gene express profiles. The script supports the annotation of four gene expression profiles, including gene-based read counts, exon-based read counts, gene-based tpm (Transcripts Per Kilobase Million), and gene-based fpkm (Fragments Per Kilobase Million). Please specify the -x option followed by the file of gene expression profiles.

<br/></li><li id='HZJACAaAAAs' class=''><b>loh_detector.pl</b> is triggered by vs_analysis.pl for the paired case/control analytic workflow. The script checks variant genotype between the affected and unaffected samples, and classifies the loss of heterozygosity (LOH) variants. Variants classified as LOH are labeled with the tag ‘is_LOH’. Further description of the paired case/control workflow is described in the next section. 

<br/></li><li id='HZJACA4egE0' class=''><b>denovo_detector.pl</b> is triggered by vs_analysis.pl for the paired case/control analytic workflow. The script compares variants both in the control and the case sample for <i>de novo</i> mutations, and labels with the tag ‘is_denovo’. Further description of the paired case/control workflow is described in the next section.

<br/></li><li id='HZJACAFQ5cO' class=''><b>Autosomal-recessive.py</b> is triggered by vs_analysis.pl for the family analytic workflow. The script filters variants of autosomal recessive inheritance. Filtered variants are labelled with the tag ‘is_AR’. 

<br/></li><li id='HZJACAQBP3O' class=''><b>Compound-het.py </b>is triggered by vs_analysis.pl for the family analytic workflow. The script filters variants of compound heterozygosity inheritance. Filtered variants are labelled with the tag ‘is_CH’. 

<br/></li><li id='HZJACAN7NC1' class=''>Denovo-recessive.py is triggered by vs_analysis.pl for the family analytic workflow. The script filters variants of <i>de novo</i> recessive mutations. Filtered variants are labelled with the tag ‘is_DR’. 

<br/></li><li id='HZJACAXLaQ2' class=''><b>Second-hit.py </b>is triggered by vs_analysis.pl for the family analytic workflow. The script filters variants of second-hit recessive mutations. Filtered variants are labelled with the tag ‘is_SH’. 

<br/></li><li id='HZJACANmHfd' class=''><b>X-linked.py </b>is triggered by vs_analysis.pl for the family analytic workflow. The script filters variants of X-chromosome linked recessive inheritance. Filtered variants are labelled with the tag ‘is_XL’. 

<br/></li></ul></ul></div><h2 id='HZJACA2d2cY'><b><b>Examples</b></b></h2>

The following examples are some common scenario.<br/>

<br/>

<h2 id='HZJACAI5rli'><b><b>Example 1 - samples of a family study</b></b></h2>

In this example, the variants on the chromosome 22 of a family trio, including NA12878, NA12891, NA12892 are used to demonstrate for family-based analysis.<br> All required files are stored in the directory ‘varselect/examples/example1/‘. <br/>

<div data-section-style='5'><ul id='HZJACAp3HY9'><li id='HZJACAvuMz4' class='' value='1'>NA12878_chr22.vcf.gz

<br/></li><li id='HZJACAVB8vf' class=''>NA12891_chr22.vcf.gz

<br/></li><li id='HZJACA2b5Ey' class=''>NA12892_chr22.vcf.gz

<br/></li><li id='HZJACAykRpL' class=''>example1.txt: A comma separated file that describes the links between the samples and the associated VCF files. 

<br/></li><li id='HZJACAqh1tk' class=''>example1.ped: A tab separated file that describes the pedigree information of this family, including gender and disease status. In this example, we hypothesize NA12878 is a sick child and that NA12891 and NA12892 are unaffected parents. The instruction of creating a ped file is available at the following website: <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped"><code>http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped</code></a>

<br/></li></ul></div>To run the analysis, please type in following command:<br/>

<pre id='HZJACAlt6gG'>varselect.pl annotate -v example1.txt  <br>                      -p example1.ped                              <br>                      -m family</pre>

 <br/>

VarSelect writes down the timestamp as analysis id when analysis begins (ex: 20170524172945), and, therefore, directory ‘VarSelectAnalysisResult_20170524172945/’ is created. The log files and the intermediate results are stored in this directory. When the original analysis is finished, ‘example1_20170524172945_varselect.db’ is created. <br/>

<br/>

Through the family workflow, five specific columns come with the analysis id: is_AR_20170524172945, is_CH_20170524172945, is_DR_20170524172945, is_SH_20170524172945 and is_XL_20170524172945 corresponding the following genetic models: autosomal recessive, compound heterozygosity, de novo recessive, second-hit recessive and X-linked recessive, respectively.<br><br/>

The following command filters the variants of compound heterozygosity inheritance and that shows the information of chromosome, position, ref allele, alt allele, gene and genotype of the selective variants.<br/>

<pre id='HZJACAXO8cp'>gemini query --header <br>             -q 'select chrom,start,ref,alt,gene,gts from variants<br>                 where is_CH_20170524172945 = 1'<br>              example1_20170524172945_varselect.db</pre>

<br> Column ‘in_analysis_20170524172945’ records variants qualifying the criteria of workflow analysis, and, in this case, is a union of the five genetic models. Variants of the five genetic models can be filtered out by following command: <br/>

<pre id='HZJACAFhVMY'>gemini query --header <br>              -q 'select chrom,start,ref,alt,is_AR_20170524172945,<br>                  is_CH_20170524172945, is_DR_20170524172945,<br>                  is_SH_20170524172945, is_XL_20170524172945<br>                  from variants where in_analysis_20170524172945=1'<br>               example1_20170524172945_varselect.db   </pre>

 <br/>

<h2 id='HZJACAA6Fli'><b>Example 2 - paired case/control samples</b></h2>

In this case, we use the variants of chromosome 22 of three samples, including blood, primary tumor ccRCC (clear cell Renal Cell Carcinoma), and a metastasis lung cancer available at <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP063388"><code>https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP063388</code></a>.<br> All files are stored in subdirectory examples/example2/<br/>

<div data-section-style='5'><ul id='HZJACAFvvoD'><li id='HZJACA13NfJ' class='' value='1'>blood-gatk-chr22.vcf.gz 

<br/></li><li id='HZJACAu9y8Q' class=''>ccRCC-gatk-chr22.vcf.gz

<br/></li><li id='HZJACAtU1s7' class=''>meta-lung-gatk-chr22.vcf.gz

<br/></li><li id='HZJACAGAMjO' class=''>example2.txt: A comma separated file that describes the  links between the sample and the corresponding VCF file.

<br/></li><li id='HZJACAX3vTX' class=''>example2.ped: A tab separated file that describes the  relationship between each sample, gender and disease status. In this case,  blood is assigned as an unaffected (control) sample, while ccRCC and  meta-lung samples are assigned as affected samples.

<br/></li></ul></div>To start the analysis, please type in the following command. <br/>

<pre id='HZJACA5upG9'> varselect.pl annotate -v example2.txt<br>                       -p example2.ped<br>                       -m paired</pre>

When the analysis is finished, the file ‘example2_20170525112231_varselect.db’ is created at the same directory. VarSelect compares the genotypic difference between the control (unaffected) and the case (affected) samples, and assign ‘LOH’ or ‘de_novo’ tags for loss of heterozygosity or <i>de novo</i> mutations. There are several columns including ‘is_LOH’, ‘LOH_samples’, ‘is_denovo’ and<b> ‘</b>denovo_samples’ specific in the paired case/control workflow. <br/>

<br> The following commands filters out the <i>de novo</i> variants.<br/>

<pre id='HZJACAlFQzp'>gemini query --header -q 'select chrom, start, ref, alt, gts<br>                          denovo_samples_20170525112231 from<br>                          variants where is_denovo_20170525112231=1'<br>                       example2_20170525112231_varselect.db</pre>

<br/>

<h2 id='HZJACAC8Toc'><b><b>Example 3 – Re-analysis and comparison (comparative analysis)</b></b></h2>

Multiple original analysis (re-analysis) can be performed for various study purposes. Comparison between any two original analyses provides the flexibility of hierarchical comparison, namely comparative analysis. User can repeat analysis by marking labels written in the ped file. For example, in the case of ccRCC, you can filter the common <i>de novo</i> mutations presented in both ccRCC and the metastatic lung samples by performing ‘blood vs. ccRCC’ and ‘blood vs. meta-lung’ analyses. The third sample in the ped file marked with '#' will be excluded in this analysis.Firstly, replicate the ped file in the sample 2 as follows.<br/>

<pre id='HZJACAwRalv'><i><i>cp example2.ped example3_mark1.ped</i></i></pre>

<br> Edit the example3_mark1.ped file and exclude meta-lung sample by marking the ‘#’sign.<br/>

<pre id='HZJACADzf1z'>example3        blood           0             0             1             1<br> example3        ccRCC           0             0             1             2<br> #example3     meta-lung          0             0             1             2</pre>

 <br/>

Perform original analysis (re-analysis) by using this new ped file. User can use ‘varselect.pl analysis’ to perform analysis and update the existing database file as the following command.<br/>

<pre id='HZJACAl2xtC'>varselect.pl analysis -d example2_20170525112231_varselect.db<br>                      -p example3_mark1.ped<br>                      -m paired</pre>

 <br/>

A new analysis id is generated at the time of beginning the re-analysis (ex: 20170525115516), and a new directory ‘VarSelectAnalysisResult_20170525115516/’ is created to store the log files and intermediate result files of this new analysis.When the re-analysis is finished, new columns tagged with new analysis id are added into the varselect_variants table.<br/>

 <br/>

User can use following command to filter out the <i>de novo</i> mutations in the ccRCC.<br/>

<pre id='HZJACAj8swg'>gemini query --header -q 'select chrom,start,ref,alt,gts from variants<br>                          where is_denovo_20170525115516 = 1'  <br>                      example2_20170525112231_varselect.db</pre>

Replicate the example3_mark1.ped file and save as the example3_mark2.ped. Mark the sample meta-lung with the ‘#’ sign. <br/>

<pre id='HZJACAEDO9J'>example3           blood         0             0             1             1<br> #example3           ccRCC         0             0             1             2<br> example3           meta-lung     0             0             1             2</pre>

 <br/>

Perform original analysis again with the new edited ped file to identify the <i>de novo</i> variants present in the metastatic lung tumor.<br/>

<pre id='HZJACA9oO8E'> varselect.pl analysis -d example2_20170525112231_varselect.db<br>                       -p example3_mark2.ped<br>                       -m paired</pre>

 <br/>

A new directory ‘VarSelectAnalysisResult_20170525121013/’ is created with the corresponding tags updated in the database.<br/>

You can extract the <i>de novo</i> mutation presented in the metastasis tumor by the following command.<br/>

<pre id='HZJACAPZRsM'>gemini query --header -q 'select chrom,start,ref,alt,gts from variants where<br>                          is_denovo_20170525121013 = 1'<br>                      example2_20170525112231_varselect.db</pre>

 <br/>

To identify the de novo mutations that present in both ccRCC and lung meta samples, we can compare the results from the two (original) analyses by intersecting the results. Please note that only analyses stored in the same database (db) file can be compared (namely comparative analysis). In this example, common <i>de novo </i>mutations can be selected by typing in the following command.<br/>

<pre id='HZJACAyk60r'>varselect.pl compare -a 20170525115516<br>                     -b 20170525121013<br>                     -c 2<br>                     -d example2_20170525112231_varselect.db</pre>

 <br/>

A new analysis id is assigned (say, 20170525145751) and the database (db) will be updated with the results of the comparative analysis. The following command lists the results stored in the database (db). <br/>

<pre id='HZJACAzguwJ'>gemini query --header -q 'select chrom, start, ref, alt, gts from variants <br>                          where in_analysis_20170525145751 = 1'<br>                      example2_20170525112231_varselect.db</pre>

 <br/>

If you want to select the novel mutations present only in the metastasis tumor, change the option –c to ‘4’ (unique to the analysis in the ‘-b’ option). <br/>

<pre id='HZJACAEdBwm'><i>varselect.</i>pl compare -a 20170525115516<br>                     -b 20170525121013   <br>                     -c 4  <br>                     -d example2_20170525112231_varselect.db</pre>

<br/>

<h2 id='HZJACAKggXg'><b><b>Example 4 – comparison of multiple variants callers</b></b></h2>

Union and/or intersection of variant calls from different variants callers could be of interests. This example demonstrates manipulation of results of two popular variant callers: ‘GATK-HaplotypeCaller’ and ‘freebayes’.<br/>

 <br/>

All files are stored in directory ‘examples/example4/’ :<br/>

<div data-section-style='5'><ul id='HZJACA2JPEz'><li id='HZJACAofYRS' class='' value='1'>NA12878-gatk-chr22.vcf.gz

<br/></li><li id='HZJACANBBYa' class=''>NA12891-gatk-chr22.vcf.gz

<br/></li><li id='HZJACAPO7x7' class=''>NA12892-gatk-chr22.vcf.gz

<br/></li><li id='HZJACAF0ZJN' class=''>NA12892-freebayes-chr22.vcf.gz: Variant calls from freebayes of chromosome 22 in these genomes

<br/></li><li id='HZJACAvaqIX' class=''>NA12878-freebayes-chr22.vcf.gz

<br/></li><li id='HZJACAn9vpt' class=''>NA12891-freebayes-chr22.vcf.gz

<br/></li><li id='HZJACAUGIpY' class=''>example4.txt: A comma separated file that describes the links between the samples and the corresponding VCF files. For multi-caller function, an extra column is required to describing the corresponding variant caller.

<br/></li></ul></div><pre id='HZJACAAmbko'>NA12878,NA12878-gatk-chr22.vcf.gz,gatk-haplotype<br> NA12891,NA12891-gatk-chr22.vcf.gz,gatk-haplotype<br> NA12892,NA12892-gatk-chr22.vcf.gz,gatk-haplotype<br> NA12878,NA12878-freebayes-chr22.vcf.gz,freebayes<br> NA12891,NA12891-freebayes-chr22.vcf.gz,freebayes<br> NA12892,NA12892-freebayes-chr22.vcf.gz,freebayes</pre>

<div data-section-style='5'><ul id='HZJACAsX4Gl'><li id='HZJACA4buUZ' class='' value='1'>example4.ped: A tab separated file that describes the relationship between each sample, gender and disease status.

<br/></li></ul></div> <br/>

The ‘-k’ option triggers the multi-caller function, followed by ‘-u’ option for preparing the union of all variants from the two variant callers. <br/>

<pre id='HZJACAtHIpg'><code>varselect.pl annotate </code><code>-v example4.txt <br></code><code>                      -p example4.ped <br>                      </code><code>-m family </code><code> <br>                      -k -u </code></pre>

A new analysis id (ex:20170524173222) is tagged on a new directory ‘VarSelectAnalysisResult_20170524173222/’. Please note that the variants with inconsistent calls by different callers are removed from the downstream analysis.<br/>

 <br/>

The full list of the removed variants could be found in the file ‘multicaller_union_inconsistant_20170524173222.txt’.<br/>

 <br/>

The following command lists the variants in this analysis.<br/>

<pre id='HZJACAgvIJS'>gemini query --header -q 'select chrom, start, ref, alt, gts from variants <br>                          where in_analysis_20170524173222 = 1'<br>                      example4_20170524173222_varselect.db</pre>

To analyze the intersection of variants identified by both variant callers, replace the option ‘-u’ to ‘-i’ in the original command as following.<br/>

<pre id='HZJACAG7dqD'>varselect.pl analysis -d example4_20170524173222_varselect.db  <br>                      -p example4.ped<br>                      -m family -k -i </pre>

Please note that the two options ‘-u’ and ‘-i’ are mutually exclusive. Inconsistent calls by different callers are removed for the downstream analysis and listed in file ‘multicaller_intersect_inconsistant_20170525103746.txt’ in the same directory of analysis output.<br/>

<h2 id='HZJACACRuNn'><b>Known issues</b></h2>

<div data-section-style='5'><ul id='HZJACALJ8aM'><li id='HZJACANQHKY' class='' value='1'>Decomposition of multiple alternative alleles on the same position results in inconsistent number of allelic quality value and that causing errors on merging VCF files. These variants are currently excluded from downstream analysis.

<br/></li><li id='HZJACAai8Pw' class=''>Variants of multiple-nucleotide substitution are incorrectly annotated as ‘indel’ by the Gemini framework. This problem only appeared in the scenario when the VCF files come from the Ion Torrent platform or converted by the CGA (Complete Genomics Analysis) tools. 

<br/></li></ul></div><br/>

<h2 id='HZJACAFpRLy'><b><b>Reference</b></b></h2>

1.        Paila U, Chapman BA, Kirchner R, Quinlan AR. GEMINI: integrative exploration of genetic variation and genome annotations. PLoS Comput Biol. 2013;9(7):e1003153. doi: 10.1371/journal.pcbi.1003153. PubMed PMID: 23874191; PubMed Central PMCID: PMCPMC3715403.<br/>

2.        Davydov EV, Goode DL, Sirota M, Cooper GM, Sidow A, Batzoglou S. Identifying a high fraction of the human genome to be under selective constraint using GERP++. PLoS Comput Biol. 2010;6(12):e1001025. doi: 10.1371/journal.pcbi.1001025. PubMed PMID: 21152010; PubMed Central PMCID: PMCPMC2996323.<br/>

3.        Kircher M, Witten DM, Jain P, O'Roak BJ, Cooper GM, Shendure J. A general framework for estimating the relative pathogenicity of human genetic variants. Nat Genet. 2014;46(3):310-5. doi: 10.1038/ng.2892. PubMed PMID: 24487276; PubMed Central PMCID: PMCPMC3992975.<br/>

4.        McLaren W, Gil L, Hunt SE, Riat HS, Ritchie GR, Thormann A, et al. The Ensembl Variant Effect Predictor. Genome Biol. 2016;17(1):122. doi: 10.1186/s13059-016-0974-4. PubMed PMID: 27268795; PubMed Central PMCID: PMCPMC4893825.<br/>

5.        Liu X, Wu C, Li C, Boerwinkle E. dbNSFP v3.0: A One-Stop Database of Functional Predictions and Annotations for Human Nonsynonymous and Splice-Site SNVs. Hum Mutat. 2016;37(3):235-41. doi: 10.1002/humu.22932. PubMed PMID: 26555599; PubMed Central PMCID: PMCPMC4752381.<br/>

6.        Liu X, Jian X, Boerwinkle E. dbNSFP: a lightweight database of human nonsynonymous SNPs and their functional predictions. Hum Mutat. 2011;32(8):894-9. doi: 10.1002/humu.21517. PubMed PMID: 21520341; PubMed Central PMCID: PMCPMC3145015.<br/>

7.        Wang K, Li M, Hakonarson H. ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data. Nucleic Acids Res. 2010;38(16):e164. doi: 10.1093/nar/gkq603. PubMed PMID: 20601685; PubMed Central PMCID: PMCPMC2938201.<br/>

8.        Cingolani P, Platts A, Wang le L, Coon M, Nguyen T, Wang L, et al. A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3. Fly (Austin). 2012;6(2):80-92. doi: 10.4161/fly.19695. PubMed PMID: 22728672; PubMed Central PMCID: PMCPMC3679285.<br/>

9.        Danecek P, Auton A, Abecasis G, Albers CA, Banks E, DePristo MA, et al. The variant call format and VCFtools. Bioinformatics. 2011;27(15):2156-8. doi: 10.1093/bioinformatics/btr330. PubMed PMID: 21653522; PubMed Central PMCID: PMCPMC3137218.<br/>

10.     Talevich E, Shain AH, Botton T, Bastian BC. CNVkit: Genome-Wide Copy Number Detection and Visualization from Targeted DNA Sequencing. PLoS Comput Biol. 2016;12(4):e1004873. doi: 10.1371/journal.pcbi.1004873. PubMed PMID: 27100738; PubMed Central PMCID: PMCPMC4839673.<br/>

11.     Anders S, Reyes A, Huber W. Detecting differential usage of exons from RNA-seq data. Genome Res. 2012;22(10):2008-17. doi: 10.1101/gr.133744.111. PubMed PMID: 22722343; PubMed Central PMCID: PMCPMC3460195.<br/>

12.     Reyes A, Anders S, Weatheritt RJ, Gibson TJ, Steinmetz LM, Huber W. Drift and conservation of differential exon usage across tissues in primate species. Proc Natl Acad Sci U S A. 2013;110(38):15377-82. doi: 10.1073/pnas.1307202110. PubMed PMID: 24003148; PubMed Central PMCID: PMCPMC3780897.<br/>

13.     Patro R, Mount SM, Kingsford C. Sailfish enables alignment-free isoform quantification from RNA-seq reads using lightweight algorithms. Nat Biotechnol. 2014;32(5):462-4. doi: 10.1038/nbt.2862. PubMed PMID: 24752080; PubMed Central PMCID: PMCPMC4077321.<br/>

14.     Liao Y, Smyth GK, Shi W. featureCounts: an efficient general purpose program for assigning sequence reads to genomic features. Bioinformatics. 2014;30(7):923-30. doi: 10.1093/bioinformatics/btt656. PubMed PMID: 24227677.<br/>

15.     Pabinger S, Dander A, Fischer M, Snajder R, Sperk M, Efremova M, et al. A survey of tools for variant analysis of next-generation genome sequencing data. Brief Bioinform. 2014;15(2):256-78. doi: 10.1093/bib/bbs086. PubMed PMID: 23341494; PubMed Central PMCID: PMCPMC3956068.<br/>

16.     Gene Ontology C. Gene Ontology Consortium: going forward. Nucleic Acids Res. 2015;43(Database issue):D1049-56. doi: 10.1093/nar/gku1179. PubMed PMID: 25428369; PubMed Central PMCID: PMCPMC4383973.<br/>

17.     Kanehisa M, Goto S. KEGG: kyoto encyclopedia of genes and genomes. Nucleic Acids Res. 2000;28(1):27-30. PubMed PMID: 10592173; PubMed Central PMCID: PMCPMC102409.<br/>

18.     Kanehisa M, Sato Y, Kawashima M, Furumichi M, Tanabe M. KEGG as a reference resource for gene and protein annotation. Nucleic Acids Res. 2016;44(D1):D457-62. doi: 10.1093/nar/gkv1070. PubMed PMID: 26476454; PubMed Central PMCID: PMCPMC4702792.<br/>

19.     Kanehisa M, Furumichi M, Tanabe M, Sato Y, Morishima K. KEGG: new perspectives on genomes, pathways, diseases and drugs. Nucleic Acids Res. 2017;45(D1):D353-D61. doi: 10.1093/nar/gkw1092. PubMed PMID: 27899662; PubMed Central PMCID: PMCPMC5210567.<br/>

 <br/>

 <br/>

<b><b> </b></b><br/>

</div>
</body></html>
